detect-nuget-inspector/detect-nuget-inspector/Inspection/Inspectors/ProjectInspector.cs

```diff
@@ -70,6 +72,69 @@ public ProjectInspector(ProjectInspectionOptions options, NugetSearchService nug
             {
                 Options.VersionName = InspectorUtil.GetProjectAssemblyVersion(Options.ProjectDirectory);
             }
+            
+            if (String.IsNullOrWhiteSpace(Options.DirectoryPackagesPropsPath))
+            {
+                Options.DirectoryPackagesPropsPath = CreateDirectoryPackagesPropsPath(Options.ProjectDirectory);
+            }
+        }
+        
+        public ProjectInspector(ProjectInspectionOptions options, NugetSearchService nugetService, HashSet<PackageId> packages, bool checkVersionOverride)
+        {
+            Options = options;
+            NugetService = nugetService;
+            if (Options == null)
+            {
+                throw new Exception("Must provide a valid options object.");
+            }
+
+            if (String.IsNullOrWhiteSpace(Options.ProjectDirectory))
+            {
+                Options.ProjectDirectory = Directory.GetParent(Options.TargetPath).FullName;
+            }
+
+            if (String.IsNullOrWhiteSpace(Options.PackagesConfigPath))
+            {
+                Options.PackagesConfigPath = CreateProjectPackageConfigPath(Options.ProjectDirectory);
+            }
+
+            if (String.IsNullOrWhiteSpace(Options.ProjectJsonPath))
+            {
+                Options.ProjectJsonPath = CreateProjectJsonPath(Options.ProjectDirectory);
+            }
+
+            if (String.IsNullOrWhiteSpace(Options.ProjectJsonLockPath))
+            {
+                Options.ProjectJsonLockPath = CreateProjectJsonLockPath(Options.ProjectDirectory);
+            }
+
+            if (String.IsNullOrWhiteSpace(Options.ProjectName))
+            {
+                Options.ProjectName = Path.GetFileNameWithoutExtension(Options.TargetPath);
+            }
+
+            if (String.IsNullOrWhiteSpace(Options.ProjectAssetsJsonPath))
+            {
+                Options.ProjectAssetsJsonPath = CreateProjectAssetsJsonPath(Options.ProjectDirectory);
+            }
+
+            if (String.IsNullOrWhiteSpace(Options.ProjectUniqueId))
+            {
+                Options.ProjectUniqueId = Path.GetFileNameWithoutExtension(Options.TargetPath);
+            }
+
+            if (String.IsNullOrWhiteSpace(Options.VersionName))
+            {
+                Options.VersionName = InspectorUtil.GetProjectAssemblyVersion(Options.ProjectDirectory);
+            }
+
+            if (String.IsNullOrWhiteSpace(Options.DirectoryPackagesPropsPath))
+            {
+                Options.DirectoryPackagesPropsPath = CreateDirectoryPackagesPropsPath(Options.ProjectDirectory);
+            }
+
+            Packages = packages;
```

Comment:
We should try to avoid code duplication by calling the preexisting constructor from this one and adding just the differences here (the same comment could be made about other classes but here the duplication is especially significant).

For a C# example, see the bottom of this page: https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/using-constructors

Reply:
Thanks, yeah that is good information, I will update all the constructors.